src/main/java/cs2103/v15_1j/jimjim/command/AddCommand.java

```diff
@@ -29,17 +31,28 @@ public AddCommand(String name, LocalDateTime start, LocalDateTime end) {
     }
 
     @Override
-    public UIFeedback undo(DataLists searchResultsList, DataLists masterList, Storage storage, Searcher searcher) {
-        // TODO
-        return null;
+    public UIFeedback undo(DataLists searchResultsList, DataLists masterList, 
+    					   Storage storage, Searcher searcher, Stack<Command> undoCommandHistory) {
+	    masterList.remove(taskEvent);
+	    
+	    if (storage.save(masterList)) {
+	        return new DeleteFeedback(taskEvent);
+	    } else {
+	        // If storage fails to save list
+	        // add task/event back to masterList and displayList
+	        masterList.add(taskEvent);
+	        undoCommandHistory.push(this);
+	        return new FailureFeedback("Some error has occured. Please try again.");
+	    }
     }
 
-    @Override
-    public UIFeedback execute(DataLists searchResultsList, DataLists masterList, Storage storage, Searcher searcher) {
+    public UIFeedback execute(DataLists searchResultsList, DataLists masterList, 
+    						  Storage storage, Searcher searcher, Stack<Command> undoCommandHistory) {
```

Comment:
I think `Stack<UndoableCommand>` makes more sense. (For the argument in undo as well). Changing this should be easy with eclipse's change method signature tool